include/boost/geometry/algorithms/detail/relate/linear_areal.hpp

```diff
@@ -1255,21 +1252,14 @@ struct linear_areal
                                                                  boost::end(range2),
                                                                  strategy);
 
+                // Calculate sides in a different point order for P and Q
                 // Will this sequence of points be always correct?
-                la_side_calculator<cs_tag, point1_type, point2_type, Strategy>
-                    side_calc(qi_conv, new_pj, pi, qi, qj, *qk_it, strategy);
-
-                return calculate_from_inside_sides(side_calc);
+                return calculate_from_inside_sides(qi, turn.point, pi, qi, qj, *qk_it, strategy);
             }
             else
             {
-                point2_type new_qj;
-                geometry::convert(turn.point, new_qj);
-
-                la_side_calculator<cs_tag, point1_type, point2_type, Strategy>
-                    side_calc(qi_conv, new_pj, pi, qi, new_qj, qj, strategy);
-
-                return calculate_from_inside_sides(side_calc);
+                // Calculate sides with different points for P and Q
+                return calculate_from_inside_sides(qi, turn.point, pi, qi, turn.point, qj, strategy);
```

Comment:
AFAIR the conversion was done because sometimes point type of one geometry is used to create a segment with the point type of another geometry. But AFAIS side strategies are able to take three different points so I think it should work but is it tested?

Reply:
Yes, side strategies should take three different point types.